Update Ray job submission example in README

dpoulopoulos opened the pull request:
Commit 0975fb2 changes the CLI groups `finetune` and `evaluate`, but the Ray job submission example in the README file still uses the deprecated `run` command.

## What's changing

Update the example to match the new API.

## How to test it

Run the example script on a Ray cluster.

veekaybee reviewed (approved):
Thanks for this!

veekaybee commented:
Will merge this on top of this PR merge, needed to pin a few dependencies https://github.com/mozilla-ai/lm-buddy/pull/119

veekaybee commented:
Related version update: https://github.com/mozilla-ai/lm-buddy/pull/120

veekaybee commented:
@dpoulopoulos Since we made a change in requirements here to satisfy failing tests, if you could also add a version bump in your PR in `pyproject.toml` so we can [satisfy the semver check](https://github.com/mozilla-ai/lm-buddy/actions/runs/10891784983/job/30223428387?pr=118), this should be good to merge!

dpoulopoulos commented:
@veekaybee I added a semver bump commit on top and rebased the whole branch on top of `main` so you can do a fast forward merge. Let me know if this is okay.

veekaybee commented:
Perfect, thanks!
